graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb

```diff
@@ -0,0 +1,159 @@
+# LinkedList data structucture that can be seen as a unidirectional graph where
+# each  vertex can have maximum two connections, and all the nodes are
+# connected "in-line" .
+class LinkedList
+  def initialize
+    @root    = nil
+    @current = nil
```

Comment:
Having a `@current` is an optimization but I would prefer that you do it only with the `@root`

Reply:
@germanescobar  I already add another implementation getting rid of current variable. PLase take a look when you can. I still missing how to deal without leaking the root variable.

Thinking out loud  Maybe linked list should only be (@val and @ref) like the basic node  but I will think about that tomorrow.